test-d/foundry/common/utils/helpers.mjs.test-d.ts

```diff
@@ -175,3 +175,111 @@ expectType<
 
 // invertObject
 expectType<{ readonly 1: 'a'; readonly foo: 'b' }>(foundry.utils.invertObject({ a: 1, b: 'foo' } as const));
+
+// mergeObject (1): tests from the docs
+expectAssignable<{ k1: string }>(foundry.utils.mergeObject({ k1: 'v1' }, { k2: 'v2' }, { insertKeys: false }));
```

Comment:
Why do you use `expectAssignable` for all the tests instead of testing more tightly with `expectType`?

Reply:
Because the first one of these fails and the second one succeeds.
```typescript
expectType<{ k1: string }>((null as unknown) as { k1: string } & Omit<{ k1: string }, keyof { k1: string }>);
expectAssignable<{ k1: string }>((null as unknown) as { k1: string } & Omit<{ k1: string }, keyof { k1: string }>);
```

Reply:
That seems very weird, it should be exactly the same type. The reason for using `expectAssignable` should at least be documented. Also it has the danger of allowing more types than needed, so we could miss some errors that we introduce. Please consider checking if there is another way and / or creating bug reports at tsd / type script for this...

Reply:
I'll look whether I find another way. The ‘issue’ is known, but no issue: https://github.com/SamVerschueren/tsd/issues/67

Reply:
Reading that issue, it sounds like it is a problem (or an intentional feature) of typescript itself. So I think it's fine to go ahead and merge this.